status_and_problems.tex

```diff
@@ -0,0 +1,134 @@
+All git packages of a software product are dependencies in that metapackage, direct or indirect.
+
+GitHub teams are used to differentiate between two types of git packages:
+
+\begin{itemize}
+\item {\bf Data Management}: it includes all dependencies of \textit{lsst\_distrib}
+\item {\bf DM Externals}: for packages used to build and distribute \textit{lsst\_distrib} but that they are not part of it.
+\end{itemize}
+
+There is a third team, {\bf DM Auxiliary}, that identify auxiliary packages to be tagged, but that they are not part of the software product:
+
+Note that in reality, \textit{lsst\_distrib} do not correspond to a software product, but to a distribution product.
+It includes not only the different pipelines, but also test data and build tools.
+
+
+\subsubsection{Environment} \label{sec:statusEnvs}
+
+The conda environment to be used for building the science pipelines is defined in the Git package \textit{scipipe\_conda\_env}.
+No releases nor tags are made  on the conda environment git package.
+
+\textit{TO BE CLARIFIED: how are java build environments managed?}
+
+
+\subsubsection{Build Tools} \label{sec:statusTools}
+
+The build tools used to build science pipelines software are available in \textit{lsstsw} and \textit{lsst\_build} git packages.
+
+The Jenkins scripts also are an important part of the tooling, facilitating a large quantity of actions, like, test a ticket branch, automate releases, or provide periodic build (weekly/daily).
+
+The tools picture is completed by \textit{codekit}, that permits to interact with multiple GitHub packages at the same time.
+Just as an example, it is used to create a tag on all the GitHub packages composing a defined product.
+
+\textit{TO BE CLARIFIED: which are java build tools?}
+
+
+\subsection{Open Problems} \label{sec:openProblems}
+
+\subsubsection{SW Product Identification} \label{sec:problemId}
+
+The definition of software product is not unique.
+
+A GitHub package may be included in more than one SW product.
+This make impossible to apply in a consistent way the release procedure to other SW products. It can be applied only to release the science pipelines.
+
+Follows an example, that explains why the procedure can't be applied to a different SW product.
+
+Product A is composed by 100 Git packages.
+
+Product B is composed by 80 Git packages.
+
+70 Git packages are shared by both products.
+
+When release 1.0 is done on product A, all 100 packages will be tagged, and corresponding Eups packages are created. This will include the 70 shared packages.
+
+One week lather, the release 1.0 should be done on product B. 
+Some of the 70 shared packages may have been updated for the need of the forthcoming 1.0 product B release.
+This imply that for some packages, release 1.0 of product A, is different from release 1.0 of product B.
+
+Requirement: each GitHub package shall be included in only one software product.
```

Comment:
I think this is in the TN somewhere already .. I think also its a recommendation rather than a requirement.   If we built every package with semantic versioning an release would depend on the hundreds of packages and that would be fine right ?  I think this is all about dependencies.

Reply:
In the definition, I say that each software product should correspond to one git repository.

I rephrase.